feat: Label and Value property for Select and MultiSelect (#1555)
appsmithorg/appsmith-docs: website/docs/reference/widgets/multiselect.md

### Data

#### Source Data `array<object>`

<dd>

> Use to set source data of the MultiSelect widget. Source Data must be specified as an array of objects. For example:

Reviewer: ```suggestion
Specify data as an array of objects to display options in the widget. For example:
```

```js
[
  {
    name: "Blue",

>     code: "BLUE",

Reviewer: ```suggestion
    code: "BLUE"
```

  },
  {
    name: "Green",
    code: "GREEN",
  },
  {
    name: "Red",

>     code: "RED",

Reviewer: ```suggestion
    code: "RED"
```

  },
];
```

> You can dynamically generate source data by fetching data from queries or JS functions and binding the response to the **Source Data** property. For example, if you have a query named `fetchData`, you can bind its response using:

Reviewer: ```suggestion
You can dynamically generate options by fetching data from queries or JS functions and binding the response to the **Source Data** property. For example, if you have a query named `fetchData`, you can bind its response using:
```

Reviewer: You also need to tell them they have to specify the label and value properties when they use query/JS

Reviewer: @Pranay105 the above change is not done. "You also need to tell them they have to specify the label and value properties when they bind using JS"

```js
{{fetchData.data}}
```

</dd>

#### Label `string`

<dd>

> Used to specify a key from the **Source Data** of the MultiSelect widget that provides the labels for each option in the list. If you want to perform some transformation using JavaScript, click the **JS** button next to the property. For example:

Reviewer: ```suggestion
Defines the key from the **Source Data** property that specifies the labels for each option in the Multiselect widget. To provide value to this field using code, click the **JS** button next to the property. For example:
```

```js

> {{ item.name.toLowerCase() }}

Reviewer: give the context for this example

```
</dd>

#### Value `string`

<dd>

> Used to define the key from the **Source Data** of the MultiSelect widget that provides the values for each option in the dropdown list. If you want to perform some transformation using JavaScript, click the **JS** button next to the property.

Reviewer: ```suggestion
Defines the key from the **Source Data** property that specifies the values for each option in the Multiselect widget. To provide value to this field using code, click the **JS** button next to the property.
```

### Data

#### Source Data `array<object>`

<dd>

Specify data as an array of objects to display options in the widget. For example:

```js
[
  {
    name: "Blue",
    code: "BLUE"
  },
  {
    name: "Green",
    code: "GREEN"
  },
  {
    name: "Red",
    code: "RED"
  },
];
```

You can dynamically generate options by fetching data from queries or JS functions and binding the response to the **Source Data** property. For example, if you have a query named `fetchData`, you can bind its response using:

```js
{{fetchData.data}}
```

> If you are generating options for MutliSelect widget using JS code as shown above, make sure to define both the [**Label**](#label-string) and [**Value**](#value-string) properties.

Reviewer: ```suggestion
If you are generating options for MutliSelect widget using JS code as shown above, you must define both the [**Label**](#label-string) and [**Value**](#value-string) properties.
```
